general/hash.hpp

```diff
@@ -210,6 +213,8 @@ class HashTable : public BlockArray<T>
    /// Check table load factor and resize if necessary
    inline void CheckRehash();
    void DoRehash();
+
+   int BinSize(int id) const;
```

Comment:
Documentation please.

Reply:
This is not a public method and besides that, what else would the docs say, other than "Returns the size of bin 'id'"?

Reply:
From the [developer guideline](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#developer-guidelines):
- All new public, protected, and private classes, methods, data members, and functions have Doxygen-style documentation in source comments.
- In addition to arguments and functionality, documentation should include the current limitations of the code, any background information that is implicitly assumed in the implementation, and the ownership and lifetime of data."

Reply:
OK...

Reply:
I believe that rule is fairly new in the guidelines. I'm all for useful comments. However, a formalistic rule that absolutely every method needs documentation does not guarantee good documentation. You will get a lot of useless comments like: 
```
/// Constructor.
ClassName() { ... }

/// Destructor.
~ClassName() { ... }

/// Returns element size.
GetElementSize() { ... }
```
Regarding other comments, please understand that I don't mean to be rude, I'm just trying to spend as little time on this as possible, since this was supposed to be a one-line bug fix, and also because I'm just a volunteer at the moment.

Reply:
I understand your point, though I don't think it applies to this case since I still have no idea what "bin" is referring to, even after reading the code. @tzanio since you agree with @jakubcerveny , do you want to write the documentation for him?

Reply:
Yohann, I don't think we need to document fundamental concepts of computer science. Please refer to https://en.wikipedia.org/wiki/Hash_table for a refresher on hash tables (they do call it "buckets" here, but other sources use "bin" for the same thing).

Reply:
Yes, it was a bug, but I did retest the experiment and the results still hold.